apis/oam/v1alpha2/core_trait_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// A ManualScalerTraitSpec defines the desired state of a ManualScalerTrait.
+type ManualScalerTraitSpec struct {
+	// ReplicaCount of the workload this trait applies to.
+	ReplicaCount int32 `json:"replicaCount"`
+
+	// WorkloadReference to the workload this trait applies to.
+	WorkloadReference WorkloadReference `json:"workloadRef"`
```

Comment:
Why do we add ref?

Reply:
It's complicated. :) See https://github.com/oam-dev/spec/issues/323.